addons/pos_pinelabs/models/pos_payment_method.py

```diff
@@ -0,0 +1,81 @@
+        response = pinelabs._call_pinelabs(endpoint='UploadBilledTransaction', payload=body)
+        if not response.get('ResponseCode') and response.get('ResponseMessage') == "APPROVED":
+            return {
+                'responseCode': response.get('ResponseCode'),
+                'status': response.get('ResponseMessage'),
+                'plutusTransactionReferenceID': response.get('PlutusTransactionReferenceID'),
+            }
+        default_error_msg = _('The expected error code for the Pine Labs POS payment upload request was not included in the response.')
+        error = response.get('ResponseMessage') or default_error_msg
+        return {'error': str(error)}
+
+    def pinelabs_fetch_payment_status(self, data):
+        pinelabs = PinelabsPosRequest(self)
+        body = pinelabs._pinelabs_request_body(payment_mode=False)
+        body.update({'PlutusTransactionReferenceID': data.get('plutusTransactionReferenceID')})
+        response = pinelabs._call_pinelabs(endpoint='GetCloudBasedTxnStatus', payload=body)
+        if response.get('ResponseCode') in [0, 1001]:
+            return {
+                'responseCode': response.get('ResponseCode'),
+                'status': response.get('ResponseMessage'),
+                'plutusTransactionReferenceID': response.get('PlutusTransactionReferenceID'),
+                'data': format_transaction_data(response.get('TransactionData')) if response.get('ResponseCode') == 0 else {},
+            }
+        default_error_msg = _('The expected error code for the Pine Labs POS payment status request was not included in the response.')
+        error = response.get('ResponseMessage') or default_error_msg
+        return {'error': str(error)}
+
+    def pinelabs_cancel_payment_request(self, data):
+        pinelabs = PinelabsPosRequest(self)
+        body = pinelabs._pinelabs_request_body(payment_mode=False)
+        body.update({
+            'Amount': data.get('amount'),
+            'PlutusTransactionReferenceID': data.get('plutusTransactionReferenceID'),
+        })
+        response = pinelabs._call_pinelabs(endpoint='CancelTransaction', payload=body)
+        if not response.get('ResponseCode') and response.get('ResponseMessage') == "APPROVED":
+            return {
+                'responseCode': response.get('ResponseCode'),
+                'error': _('Pine Labs POS transaction cancelled successfully')
+            }
+        default_error_msg = _('The expected error code for the Pine Labs POS payment cancellation request was not included in the response.')
+        errorMessage = response.get('ResponseMessage') or default_error_msg
+        return {
+            'responseCode': response.get('ResponseCode'),
+            'errorMessage': str(errorMessage)
+        }
+
+    @api.constrains('use_payment_terminal')
+    def _check_pinelabs_terminal(self):
+        if any(record.use_payment_terminal == 'pinelabs' and record.company_id.currency_id.name != 'INR' for record in self):
```

Comment:
what if the `config_id` does not have `currency_id.name == 'INR'` ?

Reply:
config_id's currency_id.name is same as record.company_id.currency_id.name. becuase config_id's currency_id computed on company_id.currency_id.name.

````
@api.depends('journal_id.currency_id', 'journal_id.company_id.currency_id', 'company_id', 'company_id.currency_id')
    def _compute_currency(self):
        for pos_config in self:
            if pos_config.journal_id:
                pos_config.currency_id = pos_config.journal_id.currency_id.id or pos_config.journal_id.company_id.sudo().currency_id.id
            else:
                pos_config.currency_id = pos_config.company_id.sudo().currency_id.id`
```

Reply:
I tested above case if config's journal_id.currency_id is different from company_id.currency_id then config will not open and log below error.

![image](https://github.com/user-attachments/assets/e2e9ffe4-a023-4d1f-8155-53c0c760443d)

Reply:
in this case we should have a `constrains` for that. Can you add it in a separate commit ?

Reply:
This may not be relevant to this pull request. We should consult MOBT about this case and create a task with appropriate cases like this.

What do you suggest?